### PR TITLE
Add Waiter iterater to encapsulate waiting for timeouts

### PR DESCRIPTION
**Pull request by vamahaja**
Add waiter class to encapsulate steps execution in loop until timeout value elapses. It aims to avoid having to write boilerplate code comparing times.

### CODE REVIEW

File: ceph/waiter.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+"""Helper object to encapsulate waiting for timeouts"""
+
+import time
+
+
+class Waiter(object):
```
Comment (psathyan) [inline ceph/waiter.py:6]: Nice. This is a good.

Could we remain the class i.e. `WaitFor` or `WaitUntil`?

**Review by psathyan — COMMENTED** [review]
Looks good to me...

I believe the module must be in `utilities` instead of `ceph`

File: ceph/waiter.py (hunk 1 above)
Comment (vamahaja) [reply ceph/waiter.py]: Changed class name to `WaitUntil`

**Review by psathyan — COMMENTED** [review]
Looks good to me... kindly provide unit test results or logs

**Comment by vamahaja** [comment]
Test Results - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-I5EDSU/

**Review by psathyan — APPROVED** [review]

**Pull request merged**